jupyter_server_proxy/handlers.py

```diff
@@ -363,7 +364,85 @@ async def proxy(self, host, port, proxied_path):
                 body = b""
             else:
                 body = None
+        accept_type = self.request.headers.get('Accept')
+        if accept_type == 'text/event-stream':
+            return await self._proxy_progressive(host, port, proxied_path, body)
+        else:
+            return await self._proxy_normal(host, port, proxied_path, body)
+    
+    async def _proxy_progressive(self, host, port, proxied_path, body):
+        # Proxy in progressive flush mode, whenever chunks are received. Potentially slower but get results quicker for voila
+
+        client = httpclient.AsyncHTTPClient()
+
+        # Set up handlers so we can progressively flush result
+
+        headers_raw = []
+
+        def dump_headers(headers_raw):
+            for line in headers_raw:
+                r = re.match('^([a-zA-Z0-9\-_]+)\s*\:\s*([^\r\n]+)[\r\n]*$', line)
```

Comment:
Can you use https://www.tornadoweb.org/en/stable/httputil.html#tornado.httputil.HTTPHeaders.parse_line here, and if that fails assume it's a status code? regexes in places like this always worry me :D

Reply:
parse_line is a internal method in HTTPHeaders, we can not use it directly

Reply:
Aaah, I see it doesn't return it but updates it. 

The regex here still worries me though. Is it coming from somewhere specific (like an RFC, or an existing implementation?)? If so, I'd appreciate a link to the source as a comment. If not, I'd like us to find some way to farm this out. I would avoid writing code that parses HTTP in any form if possible.